server/src/main/scala/com/cloudera/livy/server/interactive/InteractiveSession.scala

```diff
@@ -342,18 +342,29 @@ class InteractiveSession(
     }
   }
 
-  private def datanucleusJars(livyConf: LivyConf): Seq[String] = {
+  private def datanucleusJars(livyConf: LivyConf, sparkMajorVersion: Int): Seq[String] = {
     if (sys.env.getOrElse("LIVY_INTEGRATION_TEST", "false").toBoolean) {
       // datanucleus jars has already been in classpath in integration test
       Seq.empty
     } else {
       val sparkHome = livyConf.sparkHome().get
-      val libdir =
-        if (new File(sparkHome, "RELEASE").isFile) {
-          new File(sparkHome, "lib")
-        } else {
-          new File(sparkHome, "lib_managed/jars")
-        }
+      val libdir = sparkMajorVersion match {
+        case 1 =>
+          if (new File(sparkHome, "RELEASE").isFile) {
+            new File(sparkHome, "lib")
+          } else {
+            new File(sparkHome, "lib_managed/jars")
+          }
+        case 2 =>
+          if (new File(sparkHome, "RELEASE").isFile) {
+            new File(sparkHome, "jars")
+          } else if (new File(sparkHome, "assembly/target/scala-2.11/jars").isDirectory) {
+            new File(sparkHome, "assembly/target/scala-2.11/jars")
+          } else {
```

Comment:
Can we do a check here?
} else if (new File(sparkHome, "assembly/target/scala-2.10/jars").isDirectory) {

If all if clauses are false, can we print a warning?